modules/ROOT/pages/syntax/expressions.adoc

```diff
@@ -1406,6 +1446,21 @@ RETURN
 3+d|Rows: 1
 |===
 
+Lists can be combined with closed dynamic union types to create tests for heterogeneous lists.
+
+[source, cypher]
+----
+WITH [1, 0, true, false] AS mixedList
+RETURN booleanList IS :: LIST<BOOLEAN | INTEGER> as isMixedList
+----
```

Comment:
There's also a failing test regarding this example. 
One of `mixedList` and `booleanList` need to turn into the other :)